src/ft_messages.rs

```diff
@@ -1,45 +1,47 @@
-use ft_io::*;
-use gstd::{msg, ActorId};
+use ft_logic_io::Action;
+use ft_main_io::{FTokenAction, FTokenEvent};
+use gstd::{debug, msg, prelude::*, ActorId};
 
-pub async fn transfer_from_tokens(token_id: &ActorId, from: &ActorId, to: &ActorId, amount: u128) {
-    let _transfer_response: FTEvent = msg::send_for_reply_as(
+pub async fn transfer_tokens(
+    transaction_id: u64,
+    token_id: &ActorId,
+    sender: &ActorId,
+    recipient: &ActorId,
+    amount: u128,
+) -> Result<(), ()> {
+    let reply = msg::send_for_reply_as::<_, FTokenEvent>(
         *token_id,
-        FTAction::Transfer {
-            from: *from,
-            to: *to,
-            amount,
+        FTokenAction::Message {
+            transaction_id,
+            payload: Action::Transfer {
+                sender: *sender,
+                recipient: *recipient,
+                amount,
+            }
+            .encode(),
         },
         0,
     )
-    .unwrap()
-    .await
-    .expect("Error in transfer");
-}
+    .expect("Error in sending a message `FTokenAction::Message`")
+    .await;
 
-pub async fn transfer_tokens(token_id: &ActorId, from: &ActorId, to: &ActorId, amount: u128) {
-    let _transfer_response: FTEvent = msg::send_for_reply_as(
-        *token_id,
-        FTAction::Transfer {
-            from: *from,
-            to: *to,
-            amount,
-        },
-        0,
-    )
-    .unwrap()
-    .await
-    .expect("Error in transfer");
+    match reply {
+        Ok(FTokenEvent::Ok) => Ok(()),
+        _ => {
+            debug!("Err");
+            Err(())
+        }
+    }
 }
 
 pub async fn balance(token_id: &ActorId, account: &ActorId) -> u128 {
-    let balance_response: FTEvent =
-        msg::send_for_reply_as(*token_id, FTAction::BalanceOf(*account), 0)
-            .unwrap()
+    let reply =
+        msg::send_for_reply_as::<_, FTokenEvent>(*token_id, FTokenAction::GetBalance(*account), 0)
```

Comment:
I'd prefer to avoid turbo-fish when possible as it is not too readable.
```suggestion
    let reply: FTokenEvent =
        msg::send_for_reply_as(*token_id, FTokenAction::GetBalance(*account), 0)
```